src/crimson/os/seastore/cached_extent.h

```diff
@@ -1064,6 +1077,7 @@ class CachedExtent
       ptr = buffer_space->to_full_ptr(length);
       buffer_space.reset();
       assert(is_fully_loaded());
+      on_fully_loaded();
```

Comment:
I'm confused, shouldn't we call `on_fully_loaded()` in `on_clean_read()`? it seems that this `load_ranges` is used to get ranges to load before the actual disk read happens, so if the extent is not fully loaded before the read, this `on_fully_loaded()` can only be called when a later `read_extent()` happens. Am I missing something?

Reply:
Yeah, I admit the name `on_fully_loaded()` might be misleading, please refer to the interface comment:

```c++
  /**
   * on_fully_loaded
   *
   * Called when ptr is ready. Normally this should be used to initiate
   * the extent to be identical to CachedExtent(ptr).
   *
   * Note this doesn't mean the content is fully read, use on_clean_read for
   * this purpose.
   */
```

On the other hand, the current `on_clean_read()` serves the purpose to be called when the extent is fully read.